dttools/src/address.c

```diff
@@ -153,4 +157,27 @@ int address_parse_hostport( const char *hostport, char *host, int *port, int def
 	}
 }
 
+char *address_get_tlq_url( int port, const char *log_path ) {
```

Comment:
Some cleanup needed on this one:

1 - Instead of using raw socket code here, just use the `link` abstraction.  So, this bit of code becomes the much simpler `link_connect / link_write / link_read `.  There are a lot of technical hiccups that `link` solves for you, like ipv6, port ranges, timeouts, all sorts of things.

2 - Move this to a separate module called `tlq_config` or something like that.  The `address` module is just supposed to handle textual parsing of addresses, and is used by `link` and `datagram`, and  we want to avoid having a circular dependency.